### PR TITLE
delete taskDefinitionServiceImpl useless code

### LINKED ISSUE #13022: [Improvement][dolphinscheduler-api] delete taskDefinitionServiceImpl useless code
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

delete taskDefinitionServiceImpl useless code

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jackfanwan**
delete taskDefinitionServiceImpl useless code

closes: #13022

### CODE REVIEW

**Comment by jackfanwan** [comment]
first, i will delete useless code, decrease code complex, increase code readable
second, i will fix related bug

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java [hunk 1]
```diff
@@ -812,78 +811,89 @@ public Map<String, Object> updateTaskWithUpstream(User loginUser, long projectCo
...
+        if (result.get(Constants.STATUS) != Status.SUCCESS) {
+            return result;
         }
-        if (CollectionUtils.isNotEmpty(upstreamTaskCodes)) {
+        // update log
+        if (CollectionUtils.isNotEmpty(updateUpstreamTaskCodeSet)) {
             ProcessTaskRelation taskRelation = upstreamTaskRelations.get(0);
             List<ProcessTaskRelation> processTaskRelations =
                     processTaskRelationMapper.queryByProcessCode(projectCode, taskRelation.getProcessDefinitionCode());
-            List<ProcessTaskRelation> processTaskRelationList = Lists.newArrayList(processTaskRelations);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java:853]: It seems the `processTaskRelationList` never use.
PTAL @ruanwenjun

Comment (ruanwenjun) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java:853]: It has been used in below `for loop`.

**Comment by ruanwenjun** [comment]
I don't hope we need to refactor this class, unless you want to refactor the whole code.

**Comment by jackfanwan** [comment]
> I don't hope we need to refactor this class, unless you want to refactor the whole code.

i want refactor method updateTaskWithUpstream, because this have bug, so i want first refactor this method, then modify bug, let look simple by code

**Pull request closed without merging**